code/game/objects/effects/decals/contraband.dm

```diff
@@ -101,11 +93,12 @@
 		else
 			poster_type = pick(decls_repository.get_decl_paths_of_subtype(/decl/poster))
 	set_poster(poster_type)
+	set_dir(placement_dir)
 
 /obj/structure/sign/poster/proc/set_poster(var/poster_type)
 	var/decl/poster/design = GET_DECL(poster_type)
-	SetName("[initial(name)] - [design.name]")
-	desc = "[initial(desc)] [design.desc]"
+	SetName("poster - [design.name]")
```

Comment:
These should probably use a `base_name` and `base_desc` var like I mentioned on Discord. Hardcoding it here doesn't sit right with me.

Reply:
Why do that when all the implemented posters all prefix their name with "poster", and nothing overrides the base poster class? It's a dead-end atom.

Reply:
Because if the initial name of the base poster class ever changes, they will become out of sync. It's a redundancy issue and means you have to do twice the work.

Also, it'd be trivial to subtype it for something like flyers that can be taken down and put back up by hand, for example—I was planning on doing something like that myself.

Reply:
What will become out of sync? The "poster" string is literally used only one time in that one proc? Unless you mean the descriptions?

Also, if you want me to implement something to help you with your own PR you should just ask. That would clear things up a bit.

Reply:
> It would take about 30 seconds to add:
> 
> ```
> /decl/poster
>     var/base_name = "poster"
>     var/base_desc = "A large piece of space-resistant paper."
> ...
>     SetName("[design.base_name] - [design.name]")
>     desc = "[design.base_desc] [design.desc]"
> ```
> 
> and now the system can be trivially expanded to support placards, signs, brochures, or specific kinds of poster that aren't 'space-resistant paper'.

But signs and brochures and etc already have their own atoms? That's why I'm saying it's a dead end class. :P
I'm not trying to be a jerk or anything, I just don't like implementing things that will almost certainly never be used?

> It's really not worth arguing about and definitely not worth the implication that Haze is trying to force you to do their work for them.

That's not what I was implying. What I was saying is, instead of trying to find a reason, just ask me if you'd like me to do that change for you? That's reasonable enough I think? 
I wasn't accusing anyone here. If it was something that was actually going to be used, I'd have done it. What I didn't agree with is having to write extra features that would not get used, and eventually end up in one of those "I removed useless stuff" PR.

That's all I was trying to clarify here. Is it being added just because, or if someone is actually going to use this?

If it's that important to you I'll make the change, but I have doubts it'll ever be useful on it's own.

Reply:
Adding this a bit late, but it should be done now.